internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -821,185 +660,45 @@ func (mc *mac) handleDefaultStreamTrigger(w http.ResponseWriter, r *http.Request
-		// Just need to check the revision on one rule, use the first playbackPrefix
-		ruleKey := fmt.Sprintf("%s-%s/rule", pathKey, playbackPrefixes[0])
-		cmp := clientv3.Compare(clientv3.ModRevision(ruleKey), "=", rev.revision)
-		ctx, cancel := context.WithTimeout(context.Background(), etcdDialTimeout)
-		thn := []clientv3.Op{
-			clientv3.OpDelete(pathKey, clientv3.WithRange(pathKey+"~")),
-		}
-		if mc.baseStreamName != "" {
-			middleWaresKey := traefikKeyPathMiddlewares + etcdPlaybackID
-			thn = append(thn,
-				clientv3.OpDelete(middleWaresKey, clientv3.WithRange(middleWaresKey+"~")),
-			)
-		}
-		get := clientv3.OpGet(ruleKey)
-		resp, err := mc.etcdClient.Txn(ctx).If(cmp).Then(thn...).Else(get).Commit()
-		cancel()
-		delete(mc.etcdPub2rev, playbackID)
-		if err != nil || resp == nil {
-			glog.Errorf("mist-api-connector: error deleting keys for playbackID=%s err=%v", playbackID, err)
-		} else {
-			if resp.Succeeded {
-				glog.Errorf("mist-api-connector: success deleting keys for playbackID=%s rev=%d", playbackID, rev.revision)
-			} else {
-				var curRev int64
-				if len(resp.Responses) > 0 && len(resp.Responses[0].GetResponseRange().Kvs) > 0 {
-					curRev = resp.Responses[0].GetResponseRange().Kvs[0].CreateRevision
-				}
-				glog.Errorf("mist-api-connector: unsuccessful deleting keys for playbackID=%s myRev=%d curRev=%d pathKey=%s",
-					playbackID, rev.revision, curRev, pathKey)
-			}
-		}
-	} else {
-		glog.Errorf("mist-api-connector: etcd revision for stream playbackID=%s not found", playbackID)
-	}
-}
-
-func (mc *mac) recoverSessionLoop() {
-	if mc.etcdClient == nil {
-		return
-	}
-	clientCtx := mc.etcdClient.Ctx()
-	for clientCtx.Err() == nil {
+func (mc *mac) removeInfoDelayed(playbackID string, done chan struct{}) {
+	go func() {
 		select {
-		case <-clientCtx.Done():
-			// client closed, which means app shutted down
+		case <-done:
 			return
-		case <-mc.etcdSession.Done():
+		case <-time.After(keepStreamAfterEnd):
+			mc.removeInfo(playbackID)
 		}
-		glog.Infof("etcd session with lease=%d is lost, trying to recover", mc.etcdSession.Lease())
-
-		ctx, cancel := context.WithTimeout(clientCtx, etcdSessionRecoverTimeout)
-		err := mc.recoverEtcdSession(ctx)
-		cancel()
-
-		if err != nil && clientCtx.Err() == nil {
-			glog.Errorf("mist-api-connector: unrecoverable etcd session. err=%q.", err)
-			return
-		}
-	}
-}
-
-func (mc *mac) recoverEtcdSession(ctx context.Context) error {
-	for {
-		err := mc.recoverEtcdSessionOnce()
-		if err == nil {
-			return nil
-		}
-
-		select {
-		case <-time.After(etcdSessionRecoverBackoff):
-			glog.Errorf("mist-api-connector: Retrying etcd session recover. err=%q", err)
-			continue
-		case <-ctx.Done():
-			return fmt.Errorf("mist-api-connector: Timeout recovering etcd session err=%w", err)
-		}
-	}
+	}()
 }
 
-func (mc *mac) recoverEtcdSessionOnce() error {
-	sess, err := newEtcdSession(mc.etcdClient)
-	if err != nil {
-		return err
-	}
-
+// must be called inside mu.Lock
```

Comment:
Both functions shouldn't have this comment, correct?

Reply:
Right! Moved to only the one below.